pallets/communities/src/types.rs

```diff
@@ -0,0 +1,65 @@
+use crate::{Config, DomainNameOf};
+use frame_support::RuntimeDebug;
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_runtime::traits::Zero;
+use sp_std::{fmt, num::ParseIntError, str::FromStr, vec::Vec};
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, MaxEncodedLen, RuntimeDebug, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+#[codec(mel_bound(T: Config))]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Community<T: Config> {
+	// TODO : Maybe every community can configure an asset?
+	pub controller: T::AccountId,
+	pub population: u32,
+	pub domain_name: DomainNameOf<T>,
+}
+
+pub type BaseIndex = u8;
+pub type CategoryIndex = u8;
+pub type InstanceIndex = u16;
+
+/// CommunityId representation
+#[derive(Encode, Decode, Clone, PartialEq, Eq, MaxEncodedLen, RuntimeDebug, Default, TypeInfo)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct CommunityId {
```

Comment:
If we implement [`TypeId`](https://paritytech.github.io/substrate/master/sp_runtime/trait.TypeId.html) for `CommunityId` we get the [`AccountIdConversion`](https://paritytech.github.io/substrate/master/sp_runtime/traits/trait.AccountIdConversion.html#impl-AccountIdConversion%3CT%3E) for free.